graphics/nanovg/Portfile

```diff
@@ -0,0 +1,124 @@
+# -*- coding: utf-8; mode: tcl; tab-width: 4; indent-tabs-mode: nil; c-basic-offset: 4 -*- vim:fenc=utf-8:ft=tcl:et:sw=4:ts=4:sts=4
+
+PortSystem          1.0
+PortGroup           github 1.0
+
+set shorthash       2bead03
+set fullhash        2bead03bea43b2418060aaa154f972829995e663
+set lastcommit      20200320
+github.setup        memononen nanovg $shorthash
+version             $lastcommit
+categories          graphics
+platforms           darwin
+license             zlib
+maintainers         @jasonliu-- openmaintainer
+homepage            ${github.homepage}
+description         antialiased 2-D vector drawing library on top of \
+                    OpenGL for UI and visualizations
+long_description    NanoVG is a small antialiased vector graphics \
+                    rendering library for OpenGL. It has a lean API \
+                    modeled after the HTML5 <canvas> API. It is \
+                    intended to be a practical toolset for building \
+                    scalable user interfaces and visualizations.
+
+checksums           rmd160  b2b619156f4321b67df34e3ca8aab41e972c8328 \
+                    sha256  6467b3274d148ed550b2f148dfad0c342fa948665b15b5f3e662ae17a04443c4 \
+                    size    2032399
+
+subport metalnanovg {
+
+set shorthash       615df90
+set fullhash        615df90de566f25e25522abaadf9ea063a433a36
+set lastcommit      20200228
+github.setup        ollix MetalNanoVG ${shorthash}
+name                $subport
+version             $lastcommit
+license             MIT
+homepage            ${github.homepage}
+description         Metal port of NanoVG
+long_description    MetalNanoVG is the native Metal port of NanoVG \
+                    that tries to get the most out of Apple's Graphics \
+                    API.
+
+checksums           rmd160  e61e19a5ccb0519974080affbb88b9095f195ed3 \
+                    sha256  efc132489a075d329cc8e2e9433a016690b17efb5bfbbec5ab0ba8f2b89d5e60 \
+                    size    81691
+
+depends_lib-append  port:nanovg
+
+}
+
+# This proc is a combination of the code given at:
+#     * https://stackoverflow.com/a/435094
+#     * https://wiki.tcl-lang.org/page/recursive_glob
+# It would have been nice to use for_recursive_glob or recursive_glob
+# from the TclX package, but it seems to cause "unknown user id"
+# problems when used in a portfile.
+proc recursive_glob {dirpath patterns} {
+    set fileList {}
+    foreach fileName [ \
+        glob -nocomplain -types f -directory ${dirpath} {*}${patterns} \
+    ] {
+        lappend fileList $fileName
+    }
+    foreach dirName [ \
+        glob -nocomplain -types d -directory ${dirpath} * \
+    ] {
+        lappend fileList {*}[recursive_glob ${dirName} ${patterns}]
+    }
+    return ${fileList}
+}
+
+post-patch {
+    set files [recursive_glob ${worksrcpath}/src {*.c *.m}]
+    foreach f $files {
```

Comment:
You can replace `recursive_glob` with a simplier combination of `foreach` and `find`

```
foreach f [ exec find ${worksrcpath}/src -name "*.c" -or -name "*.m" ] { 
  ...
}
```

Reply:
Nice, this makes the file a lot cleaner. I don't know why using something like `find` didn't occur to me originally, I guess I just went with whatever was the first idea I googled.